### ISSUE TITLE
Error: Error authenticating: step[7]:

### ISSUE THREAD

**Issue by UpperCenter** [jvatic/audible-downloader#3]
Hey

I'm having trouble authenticating both on Windows (GUI) and Linux (CLI)
I get the following error message after putting in my password:
```bash
ERRO[0017] There was a problem: Enter a valid e-mail or mobile number
FATA[0017] Error: Error authenticating: step[7]: auth verification failed
```

I am using the latest release and have no issue signing in normally in a web browser. Also using the UK site if that helps.
Any help would be appreciated.

Cheers

**Comment by jvatic** [comment]
@UpperCenter thanks for the report!

I've only tested with the US and Canada sites, so it's possible you're encountering a screen the app isn't designed to handle. Can you run the app with the `LOG_LEVEL=trace` environment variable set and post the output here? (That should also output a number of html files in `$HOME/audible-downloader-config/debug` which would be useful for figuring out what's going on.)

**Comment by UpperCenter** [comment]
Thanks for the reply, how would I set that variable? I'd assume `adl --LOG_LEVEL=trace` wouldn't work.

**Comment by jvatic** [comment]
@UpperCenter Correct, `LOG_LEVEL=trace adl` should work (on Windows I think you might need to do `SET LOG_LEVEL=trace`\<enter\>, and then run `adl`.)

**Comment by UpperCenter** [comment]
Thanks, that worked. I'm slightly apprehensive about uploading logs here, as I'm not sure what they contain and might have my email address and or password in. I can see from the debug on my screen that I do get signed in successful, and the error originates here:
```bash
DEBU[0024] GET https://www.audible.co.uk/lib            
TRAC[0024] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0024] -> https://www.audible.co.uk/lib: 302 Found  
```
I can upload the actual log files, but I don't want my details public.

EDIT: That on it's own probably isn't very helpful, so here's the full logs that are displayed in the console, with sign in URLs redacted.
```bash
INFO[0003] Using www.audible.co.uk (United Kingdom)     
Audible Username: someone@example.com
Audible Password: 
DEBU[0024] getLandingPage                               
DEBU[0024] GET https://www.audible.ca/en_CA/?ipRedirectOverride=true 
TRAC[0024] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0025] -> https://www.audible.ca/en_CA/?ipRedirectOverride=true: 200 OK 
TRAC[0025] Content-Type: text/html;charset=UTF-8        
TRAC[0026] response body saved to "/home/uppercenter/audible-downloader-config/debug/01.html" 
DEBU[0026] overrideIPRedirect                           
DEBU[0026] GET https://www.audible.co.uk/en_CA/?ipRedirectOverride=true&language=en_CA 
TRAC[0026] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0026] -> https://www.audible.co.uk/en_CA/?ipRedirectOverride=true&language=en_CA: 404 Not Found 
TRAC[0026] Content-Type: text/html;charset=UTF-8        
TRAC[0026] response body saved to "/home/uppercenter/audible-downloader-config/debug/02.html" 
DEBU[0026] getSigninPage                                
DEBU[0026] GET https://www.amazon.co.uk/ap/signin?REDACTED 
TRAC[0026] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0026] -> https://www.amazon.co.uk/ap/signin?REDACTED
TRAC[0026] Content-Type: text/html;charset=UTF-8        
TRAC[0026] response body saved to "/home/uppercenter/audible-downloader-config/debug/03.html" 
DEBU[0026] doSignin                                     
DEBU[0026] POST https://www.amazon.co.uk/ap/signin/258-3474241-1849353 
TRAC[0026] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0027] -> https://www.amazon.co.uk/ap/signin/258-3474241-1849353: 200 OK 
TRAC[0027] Content-Type: text/html;charset=UTF-8        
TRAC[0027] response body saved to "/home/uppercenter/audible-downloader-config/debug/04.html" 
ERRO[0027] There was a problem: Enter a valid e-mail or mobile number 
DEBU[0027] handleCaptcha                                
DEBU[0027] handleOTPSelection                           
DEBU[0027] handleOTP                                    
DEBU[0027] confirmAuth                                  
DEBU[0027] GET https://www.audible.co.uk/lib            
TRAC[0027] User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:83.0) Gecko/20100101 Firefox/83.0 
DEBU[0027] -> https://www.audible.co.uk/lib: 302 Found  
TRAC[0027] Content-Type:                                
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/jvatic/audible-downloader/audible.logResponseBody.func1(0x0, 0x0, 0x0)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/audible/round_tripper.go:80 +0x868
github.com/sirupsen/logrus.(*Logger).LogFn(0xc0000f4000, 0xc000000006, 0xc0006824b0)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/vendor/github.com/sirupsen/logrus/logger.go:206 +0x58
github.com/sirupsen/logrus.(*Logger).TraceFn(...)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/vendor/github.com/sirupsen/logrus/logger.go:251
github.com/sirupsen/logrus.TraceFn(...)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/vendor/github.com/sirupsen/logrus/exported.go:139
github.com/jvatic/audible-downloader/audible.(*roundTripper).RoundTrip(0xbb2d38, 0xc00015a200, 0xbb2d38, 0x0, 0x0)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/audible/round_tripper.go:47 +0x6c1
net/http.send(0xc00015a200, 0x86b440, 0xbb2d38, 0x0, 0x0, 0x0, 0xc00000e188, 0x8, 0x1, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/net/http/client.go:252 +0x43e
net/http.(*Client).send(0xc000076060, 0xc00015a200, 0x0, 0x0, 0x0, 0xc00000e188, 0x0, 0x1, 0x100)
	/usr/local/Cellar/go/1.14/libexec/src/net/http/client.go:176 +0xfa
net/http.(*Client).do(0xc000076060, 0xc00015a200, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.14/libexec/src/net/http/client.go:699 +0x44a
net/http.(*Client).Do(...)
	/usr/local/Cellar/go/1.14/libexec/src/net/http/client.go:567
github.com/jvatic/audible-downloader/audible.(*Client).Do(0xc000074000, 0xc00015a200, 0x7dbeea, 0x3, 0x7dc228)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/audible/audible.go:127 +0x47
github.com/jvatic/audible-downloader/audible.(*authState).confirmAuth(0xc00000c0a0, 0x872b00, 0xc00002e080, 0x0, 0x0)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/audible/auth.go:552 +0x14a
github.com/jvatic/audible-downloader/audible.(*Client).Authenticate(0xc000074000, 0x872b00, 0xc00002e080, 0x0, 0x0)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/audible/auth.go:42 +0x287
main.(*CLI).Run(0xc000059f68, 0x872b00, 0xc0000bc440, 0xc0000bc440, 0xc000094058)
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/cli/cli.go:88 +0x5e6
main.main()
	/Users/jvatic/Workspace/go/src/github.com/jvatic/audible-downloader/cli/cli.go:35 +0x72
```

**Comment by jvatic** [comment]
Thanks @UpperCenter! It appears there's a bug in the debugging code causing the crash you see. I see what's going on with the login now though (`https://www.audible.co.uk/en_CA` doesn't exist), which appears to be caused by a URL still being hard coded. While I am fairly busy this week, I'll try and push a fix over the weekend.

**Comment by UpperCenter** [comment]
Ah that makes sense, thanks for the reply!

**Comment by jvatic** [comment]
@UpperCenter this should be fixed now in [v0.0.5](https://github.com/jvatic/audible-downloader/releases/tag/v0.0.5). Let me know if you have any further issues.
